Assets/Scripts/Menus/LoadSceneOnClick.cs

```diff
@@ -1,22 +1,50 @@
 ﻿using System.Collections;
-using System.Collections.Generic;
 using UnityEngine;
 using UnityEngine.SceneManagement;
 
+
 public class LoadSceneOnClick : MonoBehaviour
 {
-    /* Loads the scene with the given index in the project's build settings (File > Build Settings).
-     * If sceneIndex is -1, it quits the application.
+    // Spaghetti, I know
+    private AudioSource menuMusic;
+    [SerializeField] private float delay;
+
+
+    /* Loads the scene with the given name from the project's build settings (File > Build Settings).
+     * If the name is "Quit", it ends the application.
      */ 
-    public void Load(int sceneIndex)
+    public void Load(string sceneName)
     {
-        if(sceneIndex != -1)
+        if (sceneName.Equals("Menu") || sceneName.Equals("Controls"))
         {
-            SceneManager.LoadScene(sceneIndex);
+            SceneManager.LoadScene(sceneName);
         }
-        else
+        else if (sceneName.Equals("Quit"))
         {
             Application.Quit();
         }
+        else
+        {
+            menuMusic = MenuMusic.GetMusic();
+            StartCoroutine(LoadSceneDelayed(sceneName));
+        }
+    }
+
+
+    /* Delays scene transition to allow for menu music to fade.
+     */ 
```

Comment:
Not sure how I am supposed to use this method. Let's say I want to load Scene "Level2" once the player has activated my trap door. How would I take advantage of this method?

Reply:
Well, this is technically not for that, this is for menu buttons.